Automattic/jetpack
# Docker: create .env file if it doesn't exist

Makes creating `.env` file manually optional, so that docker-compose doesn't fail when the file doesn't exist.

Just a small change to smoothen any potential confusing points.

There are no plans to change env-files defined in docker-compose.yml to be optional: https://github.com/docker/compose/pull/3955#issuecomment-341231156

The file is created with Node filesystem module (https://nodejs.org/api/fs.html#fs_fs_createwritestream_path_options) to ensure cross-platform compatibility.

In Readme instructions (./docker/README.md) we already write it's optional:

> Optionally, copy settings file to modify it:
> `cp docker/default.env docker/.env`

## Testing

- Remove/backup existing .env file `rm -f docker/.env`
- Run `yarn docker:up` and observe file being created
